### ISSUE TITLE
Transport will not upgrade from polling to websocket when using SSL

### ISSUE THREAD

**Issue by nathanpeck** [socketio/socket.io-client#859]
I'm having issues with this client not upgrading the transport mechanism from polling to websocket when using an SSL socket.

The architecture is:

Web Client & iOS client -> Nginx which is terminating SSL -> Socket.io server

The iOS client can connect to the `wss://domain.com:port` address and after an initial polling request upgrades to websocket transport as expected.

However the web client connects but never upgrades, and just stays in polling mode.

As an experiment I wrote a simple node.js app using the JS client, to rule out browser issues, and the node.js app also never upgrades to websocket, and just stays in polling mode.

When I connect to a non SSL socket on the same server, it does upgrade form polling to websocket transport.

So to summarize:

SSL socket - ios socket.io client upgrades to websocket, js client does not
non SSL socket - both clients upgrade to websocket transport

Has anyone else experienced something along these lines? Is there some magic setting in the JS client that I may be missing?

Here is how we are connecting in the JS code:

```js
var ioc = require('socket.io-client');
var socket = ioc('wss://domain.com:8082', { query: 'token=' + bearerToken });
```

**Comment by driedger** [comment]
I am experiencing the same problem, using socket.io 1.3.7 on the server side, and running socket.io-client 1.3.7 in a nodejs app on the client.
Everything connects great and upgrades to a websocket properly when connecting to a non-SSL port, but when connecting over SSL, it stays on polling.

My testing is similar to whats noted in the post above, except I am using client side certificates as well (though I have tested with and without).

Currently I dont have any kind of solution or work around to get this working. Any help would be appreciated.

**Comment by driedger** [comment]
I am anxiously awaiting a solution to this problem, but after testing, it still exists in 1.4.4 as well.

Server A is an express application accepting socket.io connections over SSL/TLS. Browser clients are able to connect and immediately upgrade from polling to a websocket connection.

Server B is another node.js application, using socket.io-client to connect to server A over SSL/TLS. It connects immediately using a polling transport, but never succeeds to upgrade to a websocket. 
If I set `{transports: ['websocket']}` while connecting, it errors out immediately with  
`Error: websocket error`

**Comment by nathanpeck** [comment]
In my case the issue ended up being that the SSL certificate needed a chain
cert. Some browsers and clients trusted out cert naturally but others
required a chain in order to trust the cert. Hope this helps you.

**Comment by driedger** [comment]
Thanks, including my ca-chain cert solved my issue.

**Comment by dinhanhhuy** [comment]
@nathanpeck 
I have same error, My SSL bough on GoDaddy and test verified via this web
https://www.sslshopper.com/ssl-checker.html#hostname=google.com
Do I still need ping SSL on the iOS project?

**Comment by obipascal** [comment]
I am also having the same error in my case am using socket.io v4 both in client and server, I tried using https then including the cert and key yet it didn't work am I supposed to also use the cert on server A(server A ssl cert) which connect to Server B that accepts the socket.io connection?
